### PR TITLE
TREXIO interface

### PR DESCRIPTION
**Pull request by scemama**

### CODE REVIEW

File: ipie/utils/from_trexio.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#
+# Authors: Anthony Scemama <scemama@irsamc.ups-lse.fr>
+#
+
+"""Generate AFQMC data from a TREXIO file."""
+try:
+  import trexio
+except:
+  print("TREXIO is not installed. Try pip install trexio.")
+  raise
```
Comment (fdmalone) [inline ipie/utils/from_trexio.py:23]: raise and ImportError here

File: ipie/utils/from_trexio.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+import time
+from dataclasses import dataclass
```
Comment (fdmalone) [inline ipie/utils/from_trexio.py:26]: I know we're not currently ones to talk, but could you remove the unused imports here?

File: ipie/utils/from_trexio.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+    hcore = trexio.read_mo_1e_int_core_hamiltonian(trexio_file)
+    if verbose:
+      print(f"Read core Hamiltonian")
+
+    ndet = trexio.read_determinant_num(trexio_file)
+    if verbose:
+      print(f"{ndet} determinants")
+
+    ci_coeffs = trexio.read_determinant_coefficient(trexio_file, 0, ndet)[0]
```
Comment (fdmalone) [inline ipie/utils/from_trexio.py:69]: Does trexio use ababab ordering or aaabbb ordering for determinants? If it's ababab we might need to have to change the phase similar to what is done for dice.

File: requirements.txt [hunk 4]
```diff
@@ -3,6 +3,8 @@ h5py >= 3.0.0
 numpy >= 1.20.0
 scipy >= 1.3.0
 pytest
+pyblock
 pandas == 1.5.1 # issue with pyblock and pandas > 2
 mpi4py >= 3.0.0
-numba
\ No newline at end of file
+numba
+trexio
```
Comment (fdmalone) [inline requirements.txt:10]: Remove this as a requirement please. It should be optional.

**Review by fdmalone — CHANGES_REQUESTED** [review]
Thanks for your contribution! A unit test might be appropriate if you can supply a (small ~ 10s of kB file) which can be used as input OR (preferred) use a named temporary file to write something and read it just to check everything works. Nothing really has to be tested.

File: ipie/utils/from_trexio.py (hunk 2 above)
Comment (fdmalone) [reply ipie/utils/from_trexio.py:26]: and below too

File: requirements.txt (hunk 4 above)
Comment (linusjoonho) [reply requirements.txt:10]: Remove pyblock too, because we are not even testing that anymore.

**Comment by linusjoonho** [comment]
> @linusjoonho as a general thing do we want to add multiple interfaces to ipie?

I think so, but we should localize these interfaces in one place.
Now we have PySCF, Dice, and TREXIO. We will probably add a Q-Chem interface soon here too -
Alternatively, we can make these as separate repositories like OpenFermion does - what do you think?

**Comment by fdmalone** [comment]
Here is good, they should be single files in size so no need to complicate it.

File: ipie/utils/from_trexio.py (hunk 1 above)
Comment (scemama) [reply ipie/utils/from_trexio.py:23]: Done

File: ipie/utils/from_trexio.py (hunk 2 above)
Comment (scemama) [reply ipie/utils/from_trexio.py:26]: Done

File: ipie/utils/from_trexio.py (hunk 3 above)
Comment (scemama) [reply ipie/utils/from_trexio.py:69]: aaabbb.
we checked with Joohno, and using a CIPSI wave function coming from Quantum Package gave the correct energy for the trial wave function, and a very good estimate of the FCI energy for the AF-QMC result :-)

File: requirements.txt (hunk 4 above)
Comment (scemama) [reply requirements.txt:10]: Done

**Comment by scemama** [comment]
I will add an example in my next push.
TREXIO has 2 back-ends: HDF5 or text. The text back-end creates a directory with multiple text files. The text back-end is slower than the HDF5 one, but it is probably better suited to git repositories.

About the Q-Chem interface: It could be better to have Q-Chem export TREXIO files, because it will have more purpose than just interfacing with ipie. For example, it will automatically enable Q-Chem and Quantum Package to exchange data. So

**Comment by scemama** [comment]
man

**Comment by fdmalone** [comment]
sounds good. Just make sure the file is very small, HDF5 is probably nearly preferred as reading integrals from a textfile seems like it'll be quite slow for actual applications (not the test).

**Review by fdmalone — APPROVED** [review]

**Comment by scemama** [comment]
The example works now.

**Pull request merged**